go-controller/hybrid-overlay/pkg/controller/node_linux.go

```diff
@@ -54,26 +56,53 @@ func podToCookie(pod *kapi.Pod) string {
 	return nameToCookie(pod.Namespace + "_" + pod.Name)
 }
 
-func (n *NodeController) addOrUpdatePod(pod *kapi.Pod) error {
+func (n *NodeController) addOrUpdatePod(pod *kapi.Pod, wf *factory.WatchFactory) error {
 	podIPs, podMAC, err := getPodDetails(pod, n.nodeName)
 	if err != nil {
 		klog.V(5).Infof("cleaning up hybrid overlay pod %s/%s because %v", pod.Namespace, pod.Name, err)
 		return n.deletePod(pod)
 	}
 
+	var namespaceBackoff = wait.Backoff{Duration: 1 * time.Second, Steps: 7, Factor: 1.5, Jitter: 0.1}
+	var namespace *kapi.Namespace
+	if err = wait.ExponentialBackoff(namespaceBackoff, func() (bool, error) {
+		namespace, err = wf.GetNamespace(pod.Namespace)
+		if err != nil {
+			if errors.IsNotFound(err) {
+				// Namespace not found; don't bother waiting longer
+				return false, err
+			}
+			klog.Warningf("error getting namespace: %v", err)
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		return fmt.Errorf("failed to get namespace object: %v", err)
+	}
+	podExternalGW, ok := namespace.GetAnnotations()[hotypes.HybridOverlayExternalGw]
```

Comment:
@rcarrillocruz instead of "ok" let's use `hasHybridOverlayExtGw` or something like that, especially because it's used more than a line or two below.

Reply:
done